tests/linear/test_nearest_hahn.py

```
@@ -0,0 +1,312 @@
+from mqt.ionshuttler.linear.schedule import ActionSchedule
+from mqt.ionshuttler.linear.state import create_initial_state
+
+
+def _schedule(architecture: Architecture, path: list[Action], positions: list[int]) -> ActionSchedule:
+    return ActionSchedule.from_actions(
+        path,
+        create_initial_state(len(positions), architecture, initial_positions=positions),
+    )
+
+
+def _pulse_frame(output: DDPassResult[NearestHahnReport], architecture: Architecture, ion: int, timestep: int):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `_pulse_frame`

(ANN202)